tests/jerry/es2015/object-computed.js

```diff
@@ -0,0 +1,103 @@
+// Copyright JS Foundation and other contributors, http://js.foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/* Test object literals. */
+
+var local = 0;
+
+function f(x)
+{
+  return x + "et";
+}
+
+var o = {
+  a: 5,
+  [
+    "pr" +
+    "op"]  : 6,
+
+  [f
+   ("g")
+   ]
+   : 7,
+
+  [f(
+   "s"
+   )  ]: 8,
+
+  get    [f
+    ("res")
+    ]
+    () { return 9 },
+
+  set
+  [f("res")](
+   value) { local = value },
+};
+
+assert(o.a === 5);
+assert(o.prop === 6);
+assert(o.get === 7);
+assert(o.set === 8);
+
+local = 0;
+o.reset = 10;
+assert(local === 10);
+assert(o.reset === 9);
+
+/* Test classes. */
+
+function fxy() {
+  return "xy";
+}
+
+class C {
+  [fxy()] () {
+    return 6;
+  }
+
```

Comment:
Please add test cases for:

```js
class C {
  ["const" + "ructor"] { // 1.
    this.a = 5;
  }

  get ["constructor"]() { // 2.
    return this.a;
  }
  static ["prototype"] () { // 3.
    return 10;
  }
}
```

In the first case it must be a normal class method instead of the real constructor of the class.

The second one is a bit similar. In normal case the `get constructor() {}` is not allowed for classes (the real `constructor` cannot be an accessor), but as a computed property name it must be a simple accessor with `constructor` name.

Finally, the third one must throw a `TypeError` in runtime as the same way as `static prototype () {}` throws a `SyntaxError` during the script parsing.